baybe/kernels/base.py

```diff
@@ -0,0 +1,37 @@
+"""Base classes for all kernels."""
+
+from abc import ABC
+
+from attrs import define
+
+from baybe.serialization.core import (
+    converter,
+    get_base_structure_hook,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+from baybe.utils.basic import get_init_attributes
+
+
+@define(frozen=True)
+class Kernel(ABC, SerialMixin):
+    """Abstract base class for all kernels."""
+
+    def to_gpytorch(self, *args, **kwargs):
+        """Create the gpytorch-ready representation of the kernel."""
+        import gpytorch.kernels
+
+        kernel_cls = getattr(gpytorch.kernels, self.__class__.__name__)
+        fields_dict = get_init_attributes(
+            kernel_cls, self, ignored_keys=("self", "kwargs")
```

Comment:
why not "args" too ? (since these are the function defaults anyway why do you set them here explicitly?)

Reply:
I'm setting them explicitly since I always prefer to have stuff explicitly. Args did not show up in fields_dict, hence no reason to remove it. Could however also add this (also to the default arguments if you prefer)